chips/sam4l/src/adc.rs

```diff
@@ -393,7 +427,12 @@ impl hil::adc::Adc for Adc {
     fn sample_continuous(&self, channel: &Self::Channel, frequency: u32) -> ReturnCode {
         let regs: &mut AdcRegisters = unsafe { mem::transmute(self.registers) };
 
-        if !self.enabled.get() {
+        let res = self.config_and_enable(frequency);
```

Comment:
Not sure what you mean here, I think both `sample_continuous` and `sample_highspeed` contain this logic.